tests/test_asyncio_gsf_newcode.py

```diff
@@ -0,0 +1,569 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+#
+# The code these tests test is deprecated
+#
+# The tests remain until the code is removed
+#
+
+
+from unittest import TestCase
+import aiofiles
+from datetime import datetime
+from uuid import UUID
+
+from fixtures import async_test
+
+from mediagrains.asyncio import AsyncGSFDecoder, AsyncLazyLoaderUnloadedError, loads
+from mediagrains.grains import VideoGrain, AudioGrain, EventGrain, CodedVideoGrain, CodedAudioGrain
+from mediagrains.gsf import GSFDecodeError
+from mediagrains.gsf import GSFDecodeBadVersionError
+from mediagrains.gsf import GSFDecodeBadFileTypeError
+from mediagrains.cogenums import CogFrameFormat, CogFrameLayout, CogAudioFormat
+
+from mediatimestamp.immutable import Timestamp, TimeOffset
+
+
+with open('examples/video.gsf', 'rb') as f:
+    VIDEO_DATA = f.read()
+
+with open('examples/coded_video.gsf', 'rb') as f:
+    CODED_VIDEO_DATA = f.read()
+
+with open('examples/audio.gsf', 'rb') as f:
+    AUDIO_DATA = f.read()
+
+with open('examples/coded_audio.gsf', 'rb') as f:
+    CODED_AUDIO_DATA = f.read()
+
+with open('examples/event.gsf', 'rb') as f:
+    EVENT_DATA = f.read()
+
+with open('examples/interleaved.gsf', 'rb') as f:
+    INTERLEAVED_DATA = f.read()
+
+
+class TestAsyncGSFBlock (TestCase):
+    @async_test(suppress_warnings=True)
+    async def test_decode_headers(self):
+        async with aiofiles.open('examples/video.gsf', 'rb') as video_data_stream:
+            UUT = AsyncGSFDecoder(file_data=video_data_stream)
+            head = await UUT.decode_file_headers()
+
+        self.assertEqual(head['created'], datetime(2018, 2, 7, 10, 38, 22))
+        self.assertEqual(head['id'], UUID('163fd9b7-bef4-4d92-8488-31f3819be008'))
+        self.assertEqual(len(head['segments']), 1)
+        self.assertEqual(head['segments'][0]['id'], UUID('c6a3d3ff-74c0-446d-b59e-de1041f27e8a'))
+
+    @async_test(suppress_warnings=True)
+    async def test_generate_grains(self):
+        """Test that the generator yields each grain"""
+        async with aiofiles.open('examples/video.gsf', 'rb') as video_data_stream:
+            grain_count = 0
+            async for (grain, local_id) in AsyncGSFDecoder(file_data=video_data_stream).grains():
+                self.assertIsInstance(grain, VideoGrain)
```

Comment:
I'm not sure this file is actually testing anything beyond that `VideoGrain == VIDEOGRAIN` (et al), but since it's a type alias, that's guaranteed? You might be able to rename `test_asyncio_gsf_newcode.py` -> `test_asyncio_gsf.py`